### PR TITLE
Add Toasts to Observability Dashboards 

### PR DESCRIPTION
**Pull request by ps48**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/constants/custom_panels.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@
 export const CUSTOM_PANELS_API_PREFIX = '/api/observability/operational_panels';
 export const CUSTOM_PANELS_DOCUMENTATION_URL =
   'https://opensearch.org/docs/latest/observability-plugin/operational-panels/';
-export const CREATE_PANEL_MESSAGE = 'Enter a name to describe the purpose of this custom panel.';
+export const CREATE_PANEL_MESSAGE = 'Enter a name to describe the purpose of this observability dashboard.';
```
Comment (pjfitzgibbons) [inline common/constants/custom_panels.ts:9]: I feel we should properly capitalize the object names in messaging : "Observability Dashboard"

File: public/components/custom_panels/redux/panel_slice.ts [hunk 2]
```diff
@@ -81,6 +82,8 @@ const normalizedPanel = (panel: CustomPanelType): CustomPanelType => ({
 
 export const selectPanelList = (rootState): CustomPanelType[] => rootState.customPanel.panelList;
 
+const {setToast} = useToast();
```
Comment (pjfitzgibbons) [inline public/components/custom_panels/redux/panel_slice.ts:85]: For some reason I'm very surprised this works (?)

I think b/c in a previous incantation of `useToast` it had react state (used `setState`) and therefore was required to be inside a React-FC.  (the inception of the `use...` naming).

Glad it works outside too!

File: public/components/custom_panels/redux/panel_slice.ts [hunk 3]
```diff
@@ -136,16 +139,21 @@ export const uuidRx = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a
...
-
+    if (onSuccess) {
+      setToast(onSuccess)
+    }
     dispatch(setPanel(panel));
     const panelList = getState().customPanel.panelList.map((p) => (p.id === panel.id ? panel : p));
     dispatch(setPanelList(panelList));
-  } catch (err) {
-    console.log('Error updating Dashboard', { err, panel });
+  } catch (e) {
```
Comment (pjfitzgibbons) [inline public/components/custom_panels/redux/panel_slice.ts:152]: I'm about to believe that this try..catch block... especially the catch{} could be put in a wrapper function.
Maybe I'm over-engineering.

File: public/components/custom_panels/redux/panel_slice.ts [hunk 4]
```diff
@@ -212,40 +217,68 @@ const deleteLegacyPanels = (customPanelIdList: string[]) => {
 };
 
 export const deletePanels = (panelsToDelete: CustomPanelType[]) => async (dispatch, getState) => {
-  const ids = panelsToDelete.map((p) => p.id);
-  await Promise.all([deleteLegacyPanels(ids), deletePanelSO(ids)]);
+  const toastMessage = `Observability Dashboard${
```
Comment (pjfitzgibbons) [inline public/components/custom_panels/redux/panel_slice.ts:220]: I'm finding myself re-reading this block every time to see what it does... and that it does it right. *sigh*

Can we wrap this in a function `pluralize()` or something?

Maybe even simpler... just say "Observability Dashboard(s)".   No one truly cares... unless UX objects on some principle reason.

**Review by pjfitzgibbons — COMMENTED** [review]
I am very excited for this change!   
Thanks for taking it through to completion.

I'll approve upon responses/refactors.

File: common/constants/shared.ts [hunk 5]
```diff
@@ -25,7 +25,7 @@ export const DSL_ENDPOINT = '/_plugins/_dsl';
 
 export const observabilityID = 'observability-logs';
 export const observabilityTitle = 'Observability';
-export const observabilityPluginOrder = 6000;
+export const observabilityPluginOrder = 1500;
```
Comment (derek-ho) [inline common/constants/shared.ts:28]: what's this for?

File: common/constants/custom_panels.ts (hunk 1 above)
Comment (derek-ho) [reply common/constants/custom_panels.ts]: Done

File: public/components/custom_panels/redux/panel_slice.ts (hunk 3 above)
Comment (derek-ho) [reply public/components/custom_panels/redux/panel_slice.ts:152]: I think its over engineering for now... My 2 cents is to push it into another PR if necessary. I think this PR is already doing a lot regarding the framework around how we want to handle success and error in redux world

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**